classes/util/Loop.php

```diff
@@ -98,7 +97,10 @@ public function execute(callable $code)
         for ($i = 0; $this->looping && microtime(true) < $deadline; ++$i) {
             $result = $code();
             if (!$this->looping) { // @phpstan-ignore-line
-                break;
+                /*
+                 * The $code callback has called $this->end() and the lock has been acquired.
+                 */
```

Comment:
I couldn't help but notice that the comment style is inconsistent with the rest of the file and unnecessary for a one-line comment to use multi-line comment syntax.